bgpv2: Configuration guide for BGPv2 APIs (#32774)
cilium-team/cilium: Documentation/network/bgp-control-plane/bgp-control-plane.rst

reachability within the cluster.

.. _Border Gateway Protocol: https://datatracker.ietf.org/doc/html/rfc4271

Prerequisites
-------------

- If you are using the older MetalLB-based :ref:`bgp` feature, it must be disabled.

Installation
------------

.. tabs::

  .. group-tab:: Helm

        Cilium BGP Control Plane can be enabled with Helm flag ``bgpControlPlane.enabled``
        set as true.

        .. parsed-literal::

            $ helm upgrade cilium |CHART_RELEASE| \\
                --namespace kube-system \\
                --reuse-values \\
                --set bgpControlPlane.enabled=true
            $ kubectl -n kube-system rollout restart ds/cilium

  .. group-tab:: Cilium CLI

        .. include:: ../../installation/cli-download.rst

        Cilium BGP Control Plane can be enabled with the following command

        .. parsed-literal::

            $ cilium install |CHART_VERSION| --set bgpControlPlane.enabled=true

IPv4/IPv6 single-stack and dual-stack setup are supported. Note that the BGP
Control Plane can only advertise the route of the address family that the
Cilium is configured to use. You cannot advertise IPv4 routes when the Cilium

> Agent is configured to use only IPv6 address family. The opposite is also true.

Reviewer: I know you just copy-pasted that part, but let's take this chance to make the note very explicit:

```suggestion
Agent is configured to use only IPv6 address family. Conversely, you cannot advertise ... when ...
```

Configuring BGP Control Plane
-----------------------------

There are two ways to configure the BGP Control Plane. Using ``CiliumBGPPeeringPolicies`` resource,
or using newer BGP resources like ``CiliumBGPClusterConfig``. Currently, both configuration options are

> supported, however ``CiliumBGPPeeringPolicy`` will be deprecated in the near future.

Reviewer: ```suggestion
supported, however ``CiliumBGPPeeringPolicy`` will be deprecated in the future.
```

“near” sounds vague to me. Ideally, would you have a target Cilium version for deprecation?

Another comment: We have both `CiliumBGPPeeringPolicy` and `CiliumBGPPeeringPolicies` (plural) in this paragraph. Could we stick to just one form, to make it very clear that we're talking of the same thing?

Author: It is tentative, we are planning its removal in 1.18, and marking it deprecated in 1.17 release.

.. toctree::
    :maxdepth: 2
    :glob:

    bgp-control-plane-v1

>     bgp-control-plane-v2

Reviewer 2: should we move v2 to the top here?
